core/src/main/kotlin/com/malinskiy/marathon/report/debug/timeline/TimelineSummarySerializer.kt

```diff
@@ -0,0 +1,111 @@
+package com.malinskiy.marathon.report.debug.timeline
+
+import com.malinskiy.marathon.device.DeviceInfo
+import com.malinskiy.marathon.device.DevicePoolId
+import com.malinskiy.marathon.execution.TestResult
+import com.malinskiy.marathon.execution.TestStatus
+import com.malinskiy.marathon.report.PoolSummary
+import com.malinskiy.marathon.report.Summary
+import com.malinskiy.marathon.report.internal.TestResultSerializer
+import com.malinskiy.marathon.test.Test
+
+class TimelineSummarySerializer(private val testResultSerializer: TestResultSerializer) {
+
+    private fun prepareTestName(fullTestName: String): String {
+        return fullTestName.substring(fullTestName.lastIndexOf('.') + 1)
+    }
+
+    private fun parseData(poolId: DevicePoolId, device: DeviceInfo): List<Data> {
+        val executions = testResultSerializer.readTests(poolId, device)
+        return executions.map { this.convertToData(it) }
+    }
+
+    private fun convertToData(testResult: TestResult): Data {
+        val preparedTestName = prepareTestName(testResult.test.toString())
+        val testMetric = getTestMetric(testResult)
+        return createData(testResult, testResult.status, preparedTestName, testMetric)
+    }
+
+    private fun TestStatus.toStatus(): Status = when (this) {
```

Comment:
this logic should be synchronized with jUnit reporting. Right now it's different there.

Reply:
added support for INCOMPLETE and ASSUMPTION_FAILURE statuses